packages/protoc-gen-connect-es/src/declaration.ts

```diff
@@ -23,10 +23,12 @@ import {
 
 export function generateDts(schema: Schema) {
   for (const protoFile of schema.files) {
+    // Generate a file before checking for services to properly support
+    // keep_empty_files=true.
+    const file = schema.generateFile(protoFile.name + "_connect.d.ts");
     if (protoFile.services.length == 0) {
       continue;
     }
```

Comment:
```suggestion
    const file = schema.generateFile(protoFile.name + "_connect.d.ts");
```

Reply:
Ooh thanks for the docs update to `preamble()` — this is much simpler 🙂